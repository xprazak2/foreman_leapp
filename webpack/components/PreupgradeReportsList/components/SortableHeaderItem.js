import React from 'react';
import { Icon } from 'patternfly-react';
import classNames from 'classnames';
import PropTypes from 'prop-types';

const SortableHeaderItem = ({ value, title, sort, changeSort }) => {
  const chooseChevron = (active, order) => {
    if (!active) {
      return null;
    }

    const direction = order === 'asc' ? 'angle-up' : 'angle-down';

    return <Icon type="fa" name={direction} />;
  };

  const active = sort.attribute === value;

  return (
    <span
      className={`sortable-header ${classNames({ active })}`}
      onClick={() =>
        changeSort({
          attribute: value,
          order: sort.order === 'asc' ? 'desc' : 'asc',
        })
      }
    >
      {chooseChevron(active, sort.order)}
      {title}
    </span>
  );
};

SortableHeaderItem.propTypes = {
  sort: PropTypes.object.isRequired,
  changeSort: PropTypes.func.isRequired,
  value: PropTypes.string.isRequired,
  title: PropTypes.string.isRequired,
};

export default SortableHeaderItem;
