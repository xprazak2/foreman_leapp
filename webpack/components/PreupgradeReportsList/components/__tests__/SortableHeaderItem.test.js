import { testComponentSnapshotsWithFixtures } from '@theforeman/test';

import SortableHeaderItem from '../SortableHeaderItem';

const changeSort = () => {};

const fixtures = {
  'should render when not selected': {
    changeSort,
    sort: { attribute: '', order: 'asc' },
    value: 'severity',
    title: 'severity',
  },
  'should render when selected and sorting asc': {
    changeSort,
    sort: { attribute: 'severity', order: 'asc' },
    value: 'severity',
    title: 'Severity',
  },
  'should render when selected and sorting desc': {
    changeSort,
    sort: { attribute: 'severity', order: 'desc' },
    value: 'severity',
    title: 'Severity',
  },
};

describe('SortableHeaderItem', () =>
  testComponentSnapshotsWithFixtures(SortableHeaderItem, fixtures));
