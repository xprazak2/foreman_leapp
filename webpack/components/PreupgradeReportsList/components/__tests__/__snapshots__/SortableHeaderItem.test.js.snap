// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SortableHeaderItem should render when not selected 1`] = `
<span
  className="sortable-header "
  onClick={[Function]}
>
  severity
</span>
`;

exports[`SortableHeaderItem should render when selected and sorting asc 1`] = `
<span
  className="sortable-header active"
  onClick={[Function]}
>
  <Icon
    name="angle-up"
    type="fa"
  />
  Severity
</span>
`;

exports[`SortableHeaderItem should render when selected and sorting desc 1`] = `
<span
  className="sortable-header active"
  onClick={[Function]}
>
  <Icon
    name="angle-down"
    type="fa"
  />
  Severity
</span>
`;
